### PR TITLE
Some preparatory work to move a esbuild bundler

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: ui/webui/Makefile.am [hunk 1]
```diff
@@ -41,7 +41,27 @@ dist_noinst_DATA = \
...
+
+COCKPIT_REPO_URL = https://github.com/cockpit-project/cockpit.git
+COCKPIT_REPO_COMMIT = 6073b2703acd68e216bd9dbc116c30d2d7a9701c # 288 + esbuild-common
+
+$(COCKPIT_REPO_FILES): $(COCKPIT_REPO_STAMP)
+COCKPIT_REPO_TREE = '$(strip $(COCKPIT_REPO_COMMIT))^{tree}'
+$(COCKPIT_REPO_STAMP):
+	@git rev-list --quiet --objects $(COCKPIT_REPO_TREE) -- 2>/dev/null || \
+	     git fetch --no-tags --no-write-fetch-head --depth=1 $(COCKPIT_REPO_URL) $(COCKPIT_REPO_COMMIT)
+	cd ../../ && git archive --prefix ui/webui/ $(COCKPIT_REPO_TREE) -- $(COCKPIT_REPO_FILES) | tar x
```
Comment (jelly) [inline ui/webui/Makefile.am:62]: what about `git archive -C ../..`

**Comment by jelly** [comment]
Fails in preparation with:

```
DEBUG: make[4]: Entering directory '/builddir/build/BUILD/anaconda-39.8/ui/webui'
DEBUG: /bin/sh: line 2: git: command not found
```

File: ui/webui/Makefile.am (hunk 1 above)
Comment (KKoukiou) [reply ui/webui/Makefile.am:62]: Hmm there is not such parameter -C

Comment (jelly) [reply ui/webui/Makefile.am:62]: I guess I meant `git -C ../../ archive`

**Review by jelly — APPROVED** [review]

**Review by M4rtinK — APPROVED** [review]
Looks good to me as well - overall nice improvements in a couple places as well! :)

File: ui/webui/Makefile.am [hunk 2]
```diff
@@ -59,22 +60,34 @@ install-data-hook: $(WEBPACK_TEST)
...
+# a new project, use the latest release, and update it from time to time
+COCKPIT_REPO_FILES = \
+	pkg/lib \
+	test/common \
+	tools/git-utils.sh \
+	tools/make-bots \
+	$(NULL)
+
+COCKPIT_REPO_URL = https://github.com/cockpit-project/cockpit.git
+COCKPIT_REPO_COMMIT = 6073b2703acd68e216bd9dbc116c30d2d7a9701c # 288 + esbuild-common
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply ui/webui/Makefile.am]: @M4rtinK there is already a Jira ticket for this https://issues.redhat.com/browse/INSTALLER-2880

File: ui/webui/Makefile.am [hunk 3]
```diff
@@ -59,22 +60,34 @@ install-data-hook: $(WEBPACK_TEST)
...
+
+$(COCKPIT_REPO_FILES): $(COCKPIT_REPO_STAMP)
+COCKPIT_REPO_TREE = '$(strip $(COCKPIT_REPO_COMMIT))^{tree}'
+$(COCKPIT_REPO_STAMP): Makefile.am
+	@git rev-list --quiet --objects $(COCKPIT_REPO_TREE) -- 2>/dev/null || \
+	     git fetch --no-tags --no-write-fetch-head --depth=1 $(COCKPIT_REPO_URL) $(COCKPIT_REPO_COMMIT)
+	git -C ../../ archive $(COCKPIT_REPO_TREE) -- $(COCKPIT_REPO_FILES) | tar x
```
Comment (jkonecny12) [inline ui/webui/Makefile.am:81]: Why is this magic here. I'm somehow missing the purpose.

**Review by jkonecny12 — APPROVED** [review]

**Pull request merged**